docs/learn/expert-guides/universal-profile/extend-profile-functionalities.md

```diff
@@ -0,0 +1,555 @@
+
+Finally, test calling the tipMe function on the Universal Profile.
+
+```js
+// Attaching the ABI of TipMe to the Universal Profile so we can call tipMe() function directly on it
+const universalProfileWithExtension = new ethers.Contract(
+  universalProfileAddress,
+  TipMeABI,
+  signer,
+);
+
+// Calling the tipMe() function on the Universal Profile
+// Make sure the address associated with the private key have enough native tokens to send
+
+const tipMeTx = await universalProfileWithExtension.tipMe({
+  value: ethers.utils.parseEther('0.1'),
+});
+
+await tipMeTx.wait();
+```
+
+The event should be emitted and the value should be seen on the Universal Profile or the extension contract depending whether the value was retained or forwarded.
+
+### Final Code (Function)
+
+Below is the complete code snippet of this guide, with all the steps compiled together.
+
+```js
+import { ethers } from 'ethers';
+import { ERC725YDataKeys } from '@lukso/lsp-smart-contracts';
+
+import { abi as UniversalProfileABI } from '@lukso/lsp-smart-contracts/artifacts/UniversalProfile.json';
+
+import { abi as TipMeABI } from './TipMe.json';
+import { bytecode as TipMeBytecode } from './TipMe.json';
+
+async function main() {
+  // RPC URL (e.g., LUKSO testnet)
+  const RPC_URL = 'https://rpc.testnet.lukso.network';
+
+  // Make sure the address associated with this private key has the 'ADDEXTENSIONS' and 'CHANGEEXTENSIONS' permissions
+
+  // Replace with your private key
+  const privateKey = 'your-private-key';
+
+  const provider = new ethers.providers.JsonRpcProvider(RPC_URL);
+  const signer = new ethers.Wallet(privateKey, provider);
+
+  const universalProfileAddress = 'your-universal-profile-address';
+
+  const universalProfile = new ethers.Contract(
+    universalProfileAddress,
+    UniversalProfileABI,
+    signer,
+  );
+
+  const TipMeFactory = new ethers.ContractFactory(
+    TipMeABI,
+    TipMeBytecode,
+    signer,
+  );
+
+  const tipMeContract = await TipMeFactory.deploy();
+  await tipMeContract.deployTransaction.wait();
+
+  const tipMeInterface = new ethers.utils.Interface(TipMeABI);
+  const tipMeFunctionSelector = tipMeInterface.getSighash('tipMe()');
+
+  // Define the DataKey for the extension
+  const extensionDataKey =
+    ERC725YDataKeys.LSP17.LSP17ExtensionPrefix +
+    tipMeFunctionSelector.substr(2) +
+    '0'.repeat(32);
+
+  const tx = await universalProfile.setData(
+    extensionDataKey,
+    tipMeContract.address,
+  );
+  await tx.wait();
+
+  // Attaching the ABI of TipMe to the Universal Profile so we can call tipMe() function directly on it
+  const universalProfileWithExtension = new ethers.Contract(
+    universalProfileAddress,
+    TipMeABI,
+    signer,
+  );
+
+  // Calling the tipMe() function on the Universal Profile
+  const tipMeTx = await universalProfileWithExtension.tipMe({
+    value: ethers.utils.parseEther('0.1'),
+  });
+  await tipMeTx.wait();
+}
+
+main();
+```
+
+## Extending InterfaceIds
```

Comment:
Note sure why, but this sub-heading do not appear clear to me in the code review.

<img width="481" alt="image" src="https://github.com/lukso-network/docs/assets/31145285/c195bdc1-eabb-43a5-a02d-3b27f4cbf8ee">

Compared to this:

<img width="524" alt="image" src="https://github.com/lukso-network/docs/assets/31145285/1684459e-8fe7-4a1f-8dbb-32707af4e13f">

Suggest running `npm run start` to double check in the UI they are displayed properly.